cmd/entrypoint/snapshot_server.go

```diff
@@ -30,6 +32,26 @@ func externalSnapshotServer(ctx context.Context, snapshot *atomic.Value) error {
 			w.WriteHeader(http.StatusInternalServerError)
 			return
 		}
+		if snapDecoded.AmbassadorMeta != nil && IsEdgeStack() {
```

Comment:
See https://github.com/datawire/apro/pull/2729#discussion_r673160459 for my rationale behind this living in the snapshot_server instead of the agent.
Moreover, the `IsEdgeStack` field is not available to the agent AFAIK.